File: program_topic_go/others/order/select_sort.go
package main

func select_sort(arr []int) {
	l := len(arr)
	if l == 0 || arr == nil {
		return
	}

	min_index, tmp := 0, 0
	for i := 0; i < l; i++ {
		min_index = i
		for j := i; j < l; j++ {
			if arr[min_index] > arr[j] {
				min_index = j
			}
		}

		if min_index != i {
			tmp = arr[min_index]
			arr[min_index] = arr[i]
			arr[i] = tmp
		}
	}
}

func main() {
	arr := []int{7, 0, 1, 9, 4, 6, 3, 5, 2, 8}
	print_arr(arr)
	select_sort(arr)
	print_arr(arr)
}

//func print_arr(arr []int) {
//	l := len(arr)
//	if arr == nil || l == 0 {
//		return
//	}
//
//	for i := 0; i < l-1; i++ {
//		fmt.Print(arr[i], "\t")
//	}
//
//	fmt.Println(arr[l-1])
//}

File: program_topic_java/src/others/order/SelectSort.java
package others.order;

//简单选择排序
public class SelectSort {
    public static void main(String[] args) {
        SelectSort selectSort = new SelectSort();
        int[] arr = {8, 0, 5, 7, 9, 1, 4, 2, 6, 3};
        selectSort.printArray(arr);
        selectSort.selectSort(arr);
        selectSort.printArray(arr);
    }

    public void selectSort(int[] arr) {
        int size = arr.length;
        if (size == 0 || arr == null) {
            return;
        }

        int minIndex = 0;
        int tmp = 0;
        for (int i = 0; i < size; i++) {
            minIndex = i;
            for (int j = i; j < size; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }
            if (minIndex != i) {
                tmp = arr[minIndex];
                arr[minIndex] = arr[i];
                arr[i] = tmp;
            }
        }
    }

    public void printArray(int[] arr) {
        int size = arr.length;
        if (size == 0 || arr == null) {
            return;
        }

        for (int i = 0; i < size - 1; i++) {
            System.out.print(arr[i] + "\t");
        }
        System.out.println(arr[size - 1]);
    }
}
